src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/EventHubProcessorContext.cs

```diff
@@ -103,13 +107,18 @@ async Task OnError(ProcessErrorEventArgs arg)
 
         async Task OnMessage(ProcessEventArgs arg)
         {
-            if (!arg.HasEvent)
+            if (!arg.HasEvent || arg.CancellationToken.IsCancellationRequested || ProcessEvent == null)
                 return;
 
-            await _lockContext.Pending(arg).ConfigureAwait(false);
+            try
+            {
+                await _lockContext.Pending(arg).ConfigureAwait(false);
 
-            if (ProcessEvent != null)
-                await ProcessEvent.Invoke(arg).ConfigureAwait(false);
+                await _executorPool.Push(arg.Partition.PartitionId, () => ProcessEvent(arg), arg.CancellationToken);
```

Comment:
Random - WHY is this an `event` vs. having a strongly typed interface or delegate? Seems like a weird thing in the MassTransit codebase.